### PR TITLE
docs: Added vendor table

### PR DESCRIPTION
**Pull request by adamdbrw**
## Purpose

Clarity on open-source and alternative services used by RAI modules could be improved, as there are different vendors for text-to-speech, speech-to-text, tracing and core AI service. It is important to communicate to users what 3rd party services can be used with RAI and what could be the motivation to use them instead of open source ones, as well as linking to some helpful resources.

## Proposed Changes

Adding a neat table to summarize module vendors options.
- Improved understanding of modularity and alternatives

This PR is also adding minor improvements to other sections.

## Issues

No connected issue

## Testing

Documentation preview.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -112,24 +112,31 @@ source ./setup_shell.sh
 ### 3. Setting up vendors
 
 RAI is vendor-agnostic. Use the configuration in [config.toml](./config.toml) to set up your vendor of choice for RAI modules.
+You can skip the setup of optional modules for now. It is covered in the [Developer Guide](docs/developers_guide.md).
```
Comment (maciejmajek) [inline README.md]: What do you mean?
- Which modules are optional?
- The link doesn't work.

File: README.md [hunk 2]
```diff
@@ -112,24 +112,31 @@ source ./setup_shell.sh
...
+| AI service                 | Ollama      | OpenAI, Bedrock         | Overall performance of the AI model, supported modalities and features | [LangChain models](https://docs.langchain4j.dev/integrations/language-models/)                                                |
+| _Optional:_ Tracing tool   | Langfuse    | LangSmith               | Better integration with LangChain                                      | [Comparison](https://langfuse.com/faq/all/langsmith-alternative)                                                              |
+| _Optional:_ Text to speech | OpenTTS     | ElevenLabs              | Arguably, significantly better voice synthesis                         | <li> [OpenTTS GitHub](https://github.com/synesthesiam/opentts) </li><li> [RAI voice interface](docs/voice_interface.md) </li> |
+| _Optional:_ Speech to text | Whisper     | OpenAI Whisper (hosted) | Larger (stronger) models are available                                 | <li> [Whisper GitHub](https://github.com/openai/whisper) </li><li> [RAI voice interface](docs/voice_interface.md) </li>       |
```
Comment (maciejmajek) [inline README.md]: I think this table should be outside of the main readme. It's getting cluttered.
In terms of Whisper, I am not even sure if the hosted version offers bigger models. I actually don't think so. They offer less, but sometimes there is no choice if there is no gpu onboard for asr

File: README.md (hunk 1 above)
Comment (adamdbrw) [reply README.md]: Fixing the link. I think what is optional is well captured by the "Optional" in the module table below.

File: README.md [hunk 3]
```diff
@@ -112,24 +112,24 @@ source ./setup_shell.sh
 ### 3. Setting up vendors
 
 RAI is vendor-agnostic. Use the configuration in [config.toml](./config.toml) to set up your vendor of choice for RAI modules.
+Vendor choices for RAI are summarized in [Vendors Overview](docs/vendors_overview.md).
 
-If you do not have a vendor's key, follow the instructions below:
+For the core AI service, follow these instructions for a locally hosted Ollama [link](https://ollama.com/download).
```
Comment (maciejmajek) [inline README.md]: Can we please add information that we advise to use paid models as they are MUCH better with RAI at the moment? I did some tests, @boczekbartek did too, and as of now RAI with llama does not represent the RAI's quality at all...

**Review by maciejmajek — APPROVED** [review]
One change please

**Pull request merged**